src/cliffordsynthesis/encoding/STQGatesEncoder.cpp

```diff
@@ -0,0 +1,236 @@
+    timeStep.reserve(N);
+    for (std::size_t ctrl = 0U; ctrl < N; ++ctrl) {
+      auto& control = timeStep.emplace_back();
+      control.reserve(N + 1U);
+      for (std::size_t trgt = 0U; trgt < N; ++trgt) {
+        const std::string gName = "g_" + std::to_string(t) + "_cx_" +
+                                  std::to_string(ctrl) + "_" +
+                                  std::to_string(trgt);
+        TRACE() << "Creating variable " << gName;
+        control.emplace_back(lb->makeVariable(gName));
+      }
+
+      // Add I to the two qubit gates
+      const std::string gName = "g_" + std::to_string(t) + "_" +
+                                toString(qc::OpType::None) + "_" +
+                                std::to_string(ctrl);
+      TRACE() << "Creating variable " << gName;
+      control.emplace_back(lb->makeVariable(gName));
+    }
+  }
+}
+
+void encoding::STQGatesEncoder::collectTwoQubitGateVariables(
+    const std::size_t pos, const std::size_t qubit, const bool target,
+    LogicVector& variables) const {
+  const auto& twoQubitGates = vars.gC[pos];
+  const auto  n             = twoQubitGates.size();
+  for (std::size_t q = 0; q < n; ++q) {
+    if (q == qubit) {
+      if (N > variables.size()) {
+        variables.emplace_back(twoQubitGates[qubit].back());
+      }
+      continue;
+    }
+    if (target) {
+      variables.emplace_back(twoQubitGates[q][qubit]);
+    } else {
+      variables.emplace_back(twoQubitGates[qubit][q]);
+    }
+  }
+}
+
+void encoding::STQGatesEncoder::assertConsistency() const {
+  DEBUG() << "Asserting gate consistency";
+  for (std::size_t t = 0U; t < T / 2U; ++t) {
+    // asserting only a single gate is applied on each qubit.
+    for (std::size_t q = 0U; q < N; ++q) {
+      LogicVector singleQubitGateVariables{};
+      LogicVector twoQubitGateVariables{};
+      vars.collectSingleQubitGateVariables(t, q, singleQubitGateVariables);
+      // TODO: may need to return the vars back
+      collectTwoQubitGateVariables(t, q, true, twoQubitGateVariables);
+      collectTwoQubitGateVariables(t, q, false, twoQubitGateVariables);
+
+      IF_PLOG(plog::verbose) {
+        TRACE() << "Single Qubit Gate variables at time " << t << " and qubit "
+                << q;
+        for (const auto& var : singleQubitGateVariables) {
+          TRACE() << var.getName();
+        }
+
+        TRACE() << "Two Qubit Gate variables at time " << t << " and qubit "
+                << q;
+        for (const auto& var : twoQubitGateVariables) {
+          TRACE() << var.getName();
+        }
+      }
+
+      assertExactlyOne(singleQubitGateVariables);
+      assertExactlyOne(twoQubitGateVariables);
+    }
+  }
+}
+
+void encoding::STQGatesEncoder::assertGateConstraints() {
+  DEBUG() << "Asserting gate constraints";
+  for (std::size_t t = 0U; t < T; ++t) {
+    TRACE() << "Asserting gate constraints at time " << t;
+
+    const std::size_t pos = t < T / 2 ? t : t % (T / 2);
```

Comment:
As the Linter says, this is a possible division by Zero if `T` is 0. Also, I think it would improve clarity to add a class member that is essentially just `T/2` that is set in the constructor. Then `T` would be the number of tableaus and `T/2` would be the number of layers.